### PR TITLE
Optional formatting parameter for log

### PR DESCRIPTION
**Pull request by actioninja**
As title, for replay logging.

This is probably a horrible way of doing this, but I'm not really familiar with Rust. It works though.

### CODE REVIEW

File: src/log.rs [hunk 1]
```diff
@@ -24,15 +24,20 @@ byond_fn! { log_write(path, data) {
             Entry::Vacant(elem) => elem.insert(open(path)?),
         };
 
-        // write first line, timestamped
-        let mut iter = data.split('\n');
-        if let Some(line) = iter.next() {
-            write!(file, "[{}] {}\n", Utc::now().format("%F %T%.3f"), line)?;
-        }
+        let format_bool: bool = format.parse().unwrap();
```
Comment (Cyberboss) [inline src/log.rs]: I don't know rust so take this with a keg of salt, but isn't it possible for this to throw?

Comment (spookydonut) [inline src/log.rs]:
```suggestion
        if let Some(format_bool) = format.parse() {
```

File: src/log.rs [hunk 2]
```diff
@@ -24,15 +24,22 @@ byond_fn! { log_write(path, data) {
             Entry::Vacant(elem) => elem.insert(open(path)?),
         };
 
-        // write first line, timestamped
-        let mut iter = data.split('\n');
-        if let Some(line) = iter.next() {
-            write!(file, "[{}] {}\n", Utc::now().format("%F %T%.3f"), line)?;
-        }
+        if let Some(format_bool) = format.parse() {
```
Comment (spookydonut) [inline src/log.rs]:
```suggestion
        if let Ok(format_bool) = format.parse() {
```

Comment (spookydonut) [reply src/log.rs]: although actually you should maybe deal with a possible error from the Result

Comment (SpaceManiac) [reply src/log.rs]: suggestion: `if let Ok(false) = format.parse()` to do without formatting, to both avoid the duplicate branch and to make the old behavior the default

**Comment by actioninja** [comment]
hmm yeah ok
image

**Pull request merged**